From: Sergei Golovan
Subject: Patch adapts Snacks configure.in to the modern autoconf syntax.
Date: Fri, 15 May 2025 12:27:09 +0300

--- a/unix/configure.in
+++ b/unix/configure.in
@@ -229,7 +229,7 @@
 #--------------------------------------------------------------------
 
 # Checks for header files.
-_AC_INCLUDES_DEFAULT_REQUIREMENTS
+AC_INCLUDES_DEFAULT
 
 AC_SUBST(CC)
 AC_SUBST(CFLAGS)
@@ -263,4 +263,5 @@
 AC_SUBST(SNACK_STUB_LIB_FLAG)
 AC_SUBST(SNACK_VERSION)
 
-AC_OUTPUT(Makefile snackConfig.sh)
+AC_CONFIG_FILES([Makefile snackConfig.sh])
+AC_OUTPUT
--- a/unix/Makefile.in
+++ b/unix/Makefile.in
@@ -155,7 +155,7 @@
 	$(CC) -c $(CFLAGS) $(GENERIC_DIR)/sigproc2.c
 
 snackStubLib.o: $(GENERIC_DIR)/snackStubLib.c
-	$(CC) -c $(CFLAGS) $(GENERIC_DIR)/snackStubLib.c
+	$(CC) -c $(CFLAGS) -UALSA -UOSS $(GENERIC_DIR)/snackStubLib.c
 
 libsound${SHLIB_SUFFIX}: ${OBJSO}
 	${SHLIB_LD} ${OBJSO} ${LIBSO} -o libsound${SHLIB_SUFFIX} 
@@ -303,13 +303,13 @@
 OBJMP3 = SnackMP3.o
 
 SnackOgg.o: $(GENERIC_DIR)/SnackOgg.c
-	$(CC) @OGGINC@ -c $(CFLAGS) -DUSE_SNACK_STUBS $(GENERIC_DIR)/SnackOgg.c
+	$(CC) @OGGINC@ -c $(CFLAGS) -DUSE_SNACK_STUBS -UALSA -UOSS $(GENERIC_DIR)/SnackOgg.c
 
 libsnackogg${SHLIB_SUFFIX}: ${OBJOGG}
 	${SHLIB_LD} ${OBJOGG} ${LIBOGG} -o libsnackogg${SHLIB_SUFFIX}
 
 SnackMpg.o: $(GENERIC_DIR)/SnackMpg.c
-	$(CC) -c $(CFLAGS) -DUSE_SNACK_STUBS $(GENERIC_DIR)/SnackMpg.c
+	$(CC) -c $(CFLAGS) -DUSE_SNACK_STUBS -UALSA -UOSS $(GENERIC_DIR)/SnackMpg.c
 
 libsnackmpg${SHLIB_SUFFIX}: ${OBJMPG}
 	${SHLIB_LD} ${OBJMPG} ${LIBMPG} -o libsnackmpg${SHLIB_SUFFIX}
