From: Sergei Golovan
Subject: Patch enables build for GNU/Hurd (OSS library only).
Date: Tue, 18 Feb 2025 12:55:50 +0300

--- a/unix/configure
+++ b/unix/configure
@@ -1575,6 +1575,10 @@
 	AOBJ="jkAudIO_osx.o"
 	CFLAGS="-Os -fno-common"
         ;;
+    GNU-*)
+	AFLAG="-DOSS"
+	AOBJ="jkAudIO_oss.o"
+	;;
     *)
 	AFLAG="-DOTHER"
 	AOBJ="jkAudIO_skel.o"
--- a/unix/configure.in
+++ b/unix/configure.in
@@ -135,6 +135,10 @@
 	AOBJ="jkAudIO_osx.o"
 	CFLAGS="-Os -fno-common"
         ;;
+    GNU-*)
+	AFLAG="-DOSS"
+	AOBJ="jkAudIO_oss.o"
+	;;
     *)
 	AFLAG="-DOTHER"
 	AOBJ="jkAudIO_skel.o"
