From: Sergei Golovan
Subject: Patch fixes the makefiles to make the package buildable if
 option --shuffle is used.
Date: Fri, 16 May 2025 15:10:52 +0300
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1105691

--- a/unix/Makefile.in
+++ b/unix/Makefile.in
@@ -305,13 +305,13 @@
 SnackOgg.o: $(GENERIC_DIR)/SnackOgg.c
 	$(CC) @OGGINC@ -c $(CFLAGS) -DUSE_SNACK_STUBS -UALSA -UOSS $(GENERIC_DIR)/SnackOgg.c
 
-libsnackogg${SHLIB_SUFFIX}: ${OBJOGG}
+libsnackogg${SHLIB_SUFFIX}: ${OBJOGG} @DOSTUBLIB@
 	${SHLIB_LD} ${OBJOGG} ${LIBOGG} -o libsnackogg${SHLIB_SUFFIX}
 
 SnackMpg.o: $(GENERIC_DIR)/SnackMpg.c
 	$(CC) -c $(CFLAGS) -DUSE_SNACK_STUBS -UALSA -UOSS $(GENERIC_DIR)/SnackMpg.c
 
-libsnackmpg${SHLIB_SUFFIX}: ${OBJMPG}
+libsnackmpg${SHLIB_SUFFIX}: ${OBJMPG} @DOSTUBLIB@
 	${SHLIB_LD} ${OBJMPG} ${LIBMPG} -o libsnackmpg${SHLIB_SUFFIX}
 
 install:
